### PR TITLE
Expose `Script.get_global_name()`

### LINKED ISSUE #7474: Expose `Script.get_global_name()`
### Describe the project you are working on

This applies to a couple addons:

- A custom binary serializer
- A code generator

*There are also some other places that would benefit from a "nameof" operator, but that is a different case.*

### Describe the problem or limitation you are having in your project

I need to find the name of an arbitrary class given a reference.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Apparently this already exists in `get_global_name`, but it is not exposed to scripting. Thus, expose it to scripting.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Expose `get_global_name` to scripting.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

We can work around it for GDScript like this:

```
	var result := ""
	for global_class in ProjectSettings.get_global_class_list():
		if str(global_class["path"]) == script.resource_path:
			result = str(global_class["class"])
			break
```

<strike>However, that is only for `GDScript`. The method `get_global_name` exists in `Script`, and so it would also work for `CSharpScript` and any other future - or third party - derived class.</strike>

### Is there a reason why this should be core and not an add-on in the asset library?

The method `get_global_name` is in core.

### PR DESCRIPTION
**Pull request by theraot**
Closes https://github.com/godotengine/godot-proposals/issues/7474

### CODE REVIEW

**Comment by AThousandShips** [comment]
Needs to be documented

File: doc/classes/Script.xml [hunk 1]
```diff
@@ -30,6 +30,12 @@
 				Returns the script's base type.
 			</description>
 		</method>
+		<method name="get_global_name" qualifiers="const">
```
Comment (AThousandShips) [inline doc/classes/Script.xml:33]: These need to be in alphabetical order

**Comment by dalexeev** [comment]
I don't mind this change, but there is a related bug in GDScript:

```gdscript
class InnerClass:
    pass

func _ready():
    print((InnerClass as GDScript).get_global_name()) # InnerClass
```

https://github.com/godotengine/godot/blob/7df393312fa8eab98231f2c335366a1ef6205367/modules/gdscript/gdscript.cpp#L256-L258

https://github.com/godotengine/godot/blob/7df393312fa8eab98231f2c335366a1ef6205367/modules/gdscript/gdscript_compiler.cpp#L2928-L2930

**Review by dalexeev — APPROVED** [review]
Looks good to me, but maybe the docs could be extended.

**Comment by baileysostek** [comment]
Any movement on this? Is this a candidate for the next dev build or anything?

File: doc/classes/Script.xml [hunk 2]
```diff
@@ -24,6 +24,12 @@
 				Returns the script directly inherited by this script.
 			</description>
 		</method>
+		<method name="get_global_name" qualifiers="const">
+			<return type="StringName" />
+			<description>
+				Returns the class name associated with the script, if there is one. Returns an empty string otherwise.
```
Comment (YuriSizov) [inline doc/classes/Script.xml:30]: We should mention how you can add a global class name to a script in GDScript and C# here.

**Comment by YuriSizov** [comment]
Updated the docs and rebased on your behalf.

**Review by YuriSizov — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**